core/src/main/java/com/novoda/accessibility/ActionMenuItem.java

```diff
@@ -0,0 +1,273 @@
+        if (title != null) {
+            throw new UnsupportedOperationException();
+        }
+        return this;
+    }
+
+    @Override
+    public CharSequence getTitleCondensed() {
+        return title;
+    }
+
+    @Override
+    public ActionMenuItem setIcon(Drawable icon) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ActionMenuItem setIcon(int iconRes) {
+        if (iconRes != 0) {
+            throw new UnsupportedOperationException();
+        }
+        return this;
+    }
+
+    @Override
+    public Drawable getIcon() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ActionMenuItem setIntent(Intent intent) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Intent getIntent() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ActionMenuItem setShortcut(char numericChar, char alphaChar) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ActionMenuItem setNumericShortcut(char numericChar) {
+        Log.w("!!!", "setNumericShortcut(char) is not supported but is called by MenuInflater");
+        return this;
+    }
+
+    @Override
+    public char getNumericShortcut() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ActionMenuItem setAlphabeticShortcut(char alphaChar) {
+        Log.w("!!!", "setAlphabeticShortcut(char) is not supported but is called by MenuInflater");
+        return this;
+    }
+
+    @Override
+    public char getAlphabeticShortcut() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ActionMenuItem setCheckable(boolean checkable) {
+        if (checkable) {
+            throw new UnsupportedOperationException();
+        }
+        return this;
+    }
+
+    @Override
+    public boolean isCheckable() {
+        return false;
+    }
+
+    @Override
+    public ActionMenuItem setChecked(boolean checked) {
+        if (checked) {
+            throw new UnsupportedOperationException();
+        }
+        return this;
+    }
+
+    @Override
+    public boolean isChecked() {
+        return false;
+    }
+
+    @Override
+    public ActionMenuItem setVisible(boolean visible) {
+        if (!visible) {
+            throw new UnsupportedOperationException();
+        }
+        return this;
+    }
+
+    /**
+     * For our case, a menu item will always/only be visible if enabled.
+     */
+    @Override
+    public boolean isVisible() {
+        return isEnabled();
+    }
+
+    @Override
+    public ActionMenuItem setEnabled(boolean enabled) {
+        this.enabled = enabled;
+        return this;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    @Override
+    public boolean hasSubMenu() {
+        return false;
+    }
+
+    @Nullable
```

Comment:
I checked the `MenuItem` interface and only return null for methods where it says it's ok.